scripts/export_uml_json.py

```diff
@@ -0,0 +1,146 @@
+                definitions_header = column_names
+                print('Headers for MRDEF.RRF: {}'.format(definitions_header))
+
+    if concepts_header is None or  types_header is None or definitions_header is None:
+        print('ERROR .. column names are not found')
+        exit()
+
+    concept_details = {}  # dictionary of concept_id -> concept_dictionary
+
+    print('Reading concepts ... ')
+    with open(concepts_filename) as fin:
+        headers = concepts_header
+        for line in fin:
+            splits = line.strip().split('|')
+            assert len(headers) == len(splits)
+            concept = dict(zip(headers, splits))
+
+            if concept['LAT'] != 'ENG' or concept['SUPPRESS'] != 'N':
+                continue  # Keep English non-suppressed concepts only
+
+            # this condition is copied from S2
+            is_canonical = concept['ISPREF'] == 'Y' and concept['TS'] == 'P' and concept['STT'] == 'PF'
+
+            concept_id = concept['CUI']
+            if concept_id not in concept_details:
+                concept_details[concept_id] = {'concept_id': concept_id, 'aliases': list(), 'types': list()}
+            concept_name = concept['STR']
+
+            if not is_canonical or 'canonical_name' in concept_details[concept_id]:
+                # not a canonical name or a canonical name already found
+                concept_details[concept_id]['aliases'].append(concept_name)  # add it as an alisase
+            else:
+                concept_details[concept_id]['canonical_name'] = concept_name  # set as canonical name
+
+    print('Reading types ... ')
+    with open(types_filename) as fin:
+        headers = types_header
+        for line in fin:
+            splits = line.strip().split('|')
+            assert len(headers) == len(splits)
+            concept_type = dict(zip(headers, splits))
+
+            concept = concept_details.get(concept_type['CUI'])
+            if concept is not None:
+                concept['types'].append(concept_type['TUI'])
+
+    print('Reading definitions ... ')
+    with open(definitions_filename) as fin:
+        headers = definitions_header
+        for line in fin:
+            splits = line.strip().split('|')
+            assert len(headers) == len(splits)
+            definition = dict(zip(headers, splits))
+
+            if definition['SUPPRESS'] != 'N':
+                continue
+            is_from_preferred_source = definition['SAB'] in DEF_SOURCES_PREFERRED
+            concept = concept_details.get(definition['CUI'])
+            if concept is None:
+                continue
+
+            if 'definition' not in concept or  \
+                is_from_preferred_source and concept['is_from_preferred_source'] == 'N':
+                concept['definition'] = definition['DEF']
+                concept['is_from_preferred_source'] = 'Y' if is_from_preferred_source else 'N'
+
+    print('Number of concepts: {}'.format(len(concept_details)))
```

Comment:
Compute these statistics once in a single pass through the large dictionary, then print them out. Currently you're iterating over the whole thing 10 times unnecessarily.

As scispacy is using python 3.6, you can also use the modern formatting style for python strings:

```python
f"Here is a comment with a {variable} from the code."
```